vcd/datasource_vcd_lb_service_monitor.go

```diff
@@ -0,0 +1,90 @@
+package vcd
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func datasourceVcdLbServiceMonitor() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceVcdLbServiceMonitorRead,
+		Schema: map[string]*schema.Schema{
+			"vdc": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Edge gateway name",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Service Monitor name",
+			},
+			"interval": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
```

Comment:
Yes because datasource is meant to have set only the mandatory things for lookup org, vdc, edgegw and name. Everything else will then be returned for you.